tests/specs/lemmas.k

```diff
@@ -291,6 +291,11 @@ module LEMMAS-HASKELL [symbolic, kore]
     rule 0 <=Int #sizeByteArray ( _ )     => true [smt-lemma]
     rule 0 <=Int lengthBytes ( _ )        => true [smt-lemma]
 
-    rule bool2Word( B:Bool ) ==Int 1 => B         [simplification]
-    rule bool2Word( B:Bool ) ==Int 0 => notBool B [simplification]
+    rule I ==Int bool2Word( B:Bool ) => bool2Word(B) ==Int I            [simplification, concrete(I)]
+    rule bool2Word( B:Bool ) ==Int I => B ==K word2Bool(I)              [simplification, concrete(I)]
+
+    rule BA:ByteArray ==K #buf( 32, DATA ) => #buf( 32, DATA ) ==K BA   [simplification, concrete(BA)]
+    rule #buf( 32, DATA ) ==K BA:ByteArray => DATA ==Int #asInteger(BA)
+      requires #sizeByteArray(BA) <=Int 32                              [simplification, concrete(BA)]
```

Comment:
I think you may also need `0 <=Int DATA andBool DATA <Int pow256` here, because otherwise `#buf` is ill-defined.

Reply:
And if `#buf` is ill-defined, we shouldn't get `false` here, we should get `#False` (`\bottom`).

Reply:
This is the job of `#Ceil(#buf)` which I disabled for now because it's prohibitive for performance. If we add it here we'll likely forget it and have lingering performance issues.

Reply:
@denis-bogdanas I don't understand "disabled for now" here? I still see this `#Ceil` condition in `edsl.md`, and it looks like it's still enabled.

I think the only reason this is sound without the additional range checks on `DATA` is because of that `#Ceil` condition, right?

Reply:
Right. That's why we don't need extra checks here. It is disabled in symbolic testing branches.